submissions/sofiichuk/oop-exercise/index.js

```diff
@@ -0,0 +1,75 @@
+class Animal {
+    constructor(name, gender, saying, legs) {
+        this.name = name;
+        this.gender = gender;
+        this.saying = saying;
+        this.legs = legs;
+    }
+
+    declareAnimal() {
+        return [this.name, this.gender, this.saying, this.legs].join(";");
+    }
+}
+
+class Human extends Animal {
+    constructor(name, gender, saying, legs = 2, arms = 2) {
+        super(name, gender, saying, legs);
+        this.arms = arms;
+    }
+
+    declareGenus() {
+        return [this.constructor.name, super.declareAnimal(), this.arms].join(";");
+    }
+}
+
+class Pet extends Animal {
+    constructor(name, gender, saying, legs = 4) {
+        super(name, gender, saying, legs);
+    }
+
+    declareGenus() {
+        return [this.constructor.name, super.declareAnimal()].join(";");
+    }
```

Comment:
`declareGenus` rather represents class instance.
It is OK to give the same name to the methods that have basically the same purpose in child classes.
`super.methodName` will call inherited version anyway.